src/marqo/tensor_search/tensor_search.py

```diff
@@ -1235,6 +1346,257 @@ def _lexical_search(
+
+    ## 2. Vectorise against all queries
+    job_ptr_to_vectors: Dict[JHash, List[float]] = {}
+    for idx, v in jobs.items():
+        job_ptr_to_vectors[idx] = functools.reduce(
+            lambda x, y: x + y,
+            [s2_inference.vectorise(
+                model_name=v.model_name, model_properties=v.model_properties,
+                content=batch, device=v.device,
+                normalize_embeddings=v.normalize_embeddings,
+                image_download_headers=v.image_download_headers
+            ) for batch in v.content] # Todo: check if should be batching.
+        )
+
+    ## 3. Get vectors back from batch, process `ordered_queries` into each query.
+    qidx_to_vectors: Dict[Qidx, List[List[float]]] = defaultdict(list)
+    for qidx, ptr in qidx_to_job.items():
+        vectors = job_ptr_to_vectors[ptr.job_hash][ptr.start_idx: ptr.end_idx]
+        qidx_to_vectors[qidx].append(vectors)
+
+        q = queries[qidx]
+        ordered_queries = list(q.q.items()) if isinstance(q.q, dict) else None
+        if ordered_queries:
+            # multiple queries. We have to weight and combine them:
+            index_info = get_index_info(config=config, index_name=q.index)
+            weighted_vectors = [np.asarray(vec) * weight for vec, weight in zip(vectors, [w for _, w in ordered_queries])]
+            vectorised_text = np.mean(weighted_vectors, axis=0)
+            if index_info.index_settings['index_defaults']['normalize_embeddings']:
+                norm = np.linalg.norm(vectorised_text, axis=-1, keepdims=True)
+                if norm > 0:
+                    vectorised_text /= np.linalg.norm(vectorised_text, axis=-1, keepdims=True)
+            qidx_to_vectors[qidx] = list(vectorised_text)
+        else:
+            qidx_to_vectors[qidx] = vectors[0]
+
+    ## 4. Create msearch request bodies and combine to aggregate.
+    query_to_body_parts: Dict[Qidx, List[Dict]] = {}
+    query_to_body_count: Dict[Qidx, int] = {} # Keep track of count, so we can separate after msearch call.
+    for qidx, qq in enumerate(queries):
+        index_info = get_index_info(config=config, index_name=q.index)
+        if q.filter is not None:
+            contextualised_filter = utils.contextualise_filter(
+                filter_string=q.filter,
+                simple_properties=index_info.get_text_properties())
+        else:
+            contextualised_filter = ''
+
+        vector_properties_to_search = get_vector_properties_to_search(qq.searchableAttributes, index_info, raise_on_searchable_attribs)
+        vectorised_text = qidx_to_vectors[qidx]
+        body = construct_msearch_body_elements(vector_properties_to_search, qq.offset, q.filter, index_info, qq.limit, vectorised_text, qq.attributesToRetrieve, qq.index, contextualised_filter)
+        query_to_body_parts[qidx] = body
+        query_to_body_count[qidx] = len(body)
+
+    aggregate_body = [line for body in query_to_body_parts.values() for line in body]
```

Comment:
what does this do?

Reply:
is this a reduce? If so I think it is more readable as an explicit `reduce()` call

Reply:
This is a flattening of first order. e.g.: 
```
[[{}, {}, {}, {}], [{}, {}, {}], [{}, {}, {}, {}], [{}, {}]]
```
becomes
```
[{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
```

Reply:
nit: to me this is more readable `functools.reduce(lambda x, y: x + y, query_to_body_parts.values())`

Reply:
Maybe add a comment above it saying this flattens it